packages/ui-text-area/src/TextArea/styles.ts

```diff
@@ -72,54 +80,61 @@ const generateStyle = (
       fontFamily: componentTheme.fontFamily,
       fontWeight: componentTheme.fontWeight,
       borderWidth: componentTheme.borderWidth,
-      borderStyle: componentTheme.borderStyle,
-      borderTopColor: componentTheme.borderTopColor,
-      borderRightColor: componentTheme.borderRightColor,
-      borderBottomColor: componentTheme.borderBottomColor,
-      borderLeftColor: componentTheme.borderLeftColor,
+      borderStyle: 'solid',
+      borderColor: componentTheme.borderColor,
       borderRadius: componentTheme.borderRadius,
-      color: componentTheme.color,
-      background: componentTheme.background,
+      color: componentTheme.textColor,
+      background: componentTheme.backgroundColor,
       whiteSpace: 'pre-wrap',
       wordWrap: 'break-word',
       textAlign: 'start',
+      ...(!disabled &&
+        !readOnly && {
+          '&:hover': {
+            color: componentTheme.textHoverColor,
+            background: componentTheme.backgroundHoverColor,
+            borderColor: componentTheme.borderHoverColor
+          }
+        }),
+      ...(readOnly && {
+        color: componentTheme.textReadonlyColor,
+        background: componentTheme.backgroundReadonlyColor,
+        borderColor: componentTheme.borderReadonlyColor
+      }),
+      ...(disabled && {
+        color: componentTheme.textDisabledColor,
+        background: componentTheme.backgroundDisabledColor,
+        borderColor: componentTheme.borderDisabledColor,
+        cursor: 'not-allowed',
+        pointerEvents: 'none',
+        opacity: 0.5
+      }),
+      ...(success && {
+        borderColor: componentTheme.successBorderColor
+      }),
       ...fontSizeVariants[size!],
-      '&:focus + [class$=-textArea__outline]': {
-        transform: 'scale(1)',
-        opacity: 1
-      },
-      '&[aria-invalid], &[aria-invalid]:focus, &[aria-invalid]:focus + [class$=-textArea__outline]': {
+      ...calcFocusOutlineStyles(sharedTokens.focusOutline, {
+        focusColor
+      }),
+      '&[aria-invalid], &[aria-invalid]:focus': {
         borderColor: componentTheme.errorBorderColor
       },
```

Comment:
the border should be grey when disabled/readonly

Reply:
oh I see. So we are ONLY showing info messages for these, not `error` and `success`